app/model/CampaignReferral.scala

```diff
@@ -20,11 +20,29 @@ object Component {
 // A referral to any item in a campaign from an on-platform source in a particular period
 case class CampaignReferral(
   component: Component,
-  numClicks: Int,
+  clickCount: Int,
+  impressionCount: Int,
+  // Annoyingly react won't recognise CTR if it's a derived property so it has to go in as part of the constructor
```

Comment:
I don't understand what you mean here, can you explain please?

Reply:
Thanks for questioning it.  I realise that it makes no sense at all, and I've fixed the underlying problem now.

Reply:
Great :)